src/app/DIAmeterApplication.cpp

```diff
@@ -553,6 +1409,67 @@ COMMAND_T DIAmeterApplication::getCommand() const {
 }
 
 void DIAmeterApplication::processParams() {
+	const string index = Params::GetString("tide database");
+	if (!FileUtils::Exists(index)) { carp(CARP_FATAL, "'%s' does not exist", index.c_str()); }
+	else if (FileUtils::IsRegularFile(index)) { // Index is FASTA file
+		carp(CARP_INFO, "Creating index from '%s'", index.c_str());
+		string targetIndexName = Params::GetString("store-index");
+		if (targetIndexName.empty()) {
+			targetIndexName = FileUtils::Join(Params::GetString("output-dir"), "tide-search.tempindex");
+		}
+		remove_index_ = targetIndexName;
+
+		Params::Set("peptide-list", true);
+		Params::Set("decoy-format", "peptide-reverse");
+
+		TideIndexApplication indexApp;
+		indexApp.processParams();
+		if (indexApp.main(index, targetIndexName) != 0) { carp(CARP_FATAL, "tide-index failed."); }
+		Params::Set("tide database", targetIndexName);
+
+	} else { // Index is Tide index directory
+		pb::Header peptides_header;
+		string peptides_file = FileUtils::Join(index, "pepix");
+		HeadedRecordReader peptide_reader(peptides_file, &peptides_header);
+		if ((peptides_header.file_type() != pb::Header::PEPTIDES) || !peptides_header.has_peptides_header()) { carp(CARP_FATAL, "Error reading index (%s).", peptides_file.c_str()); }
+
+		const pb::Header::PeptidesHeader& pepHeader = peptides_header.peptides_header();
+
+		Params::Set("enzyme", pepHeader.enzyme());
+		const char* digestString = digest_type_to_string(pepHeader.full_digestion() ? FULL_DIGEST : PARTIAL_DIGEST);
+		Params::Set("digestion", digestString);
+		Params::Set("isotopic-mass", pepHeader.monoisotopic_precursor() ? "mono" : "average");
+	}
+
+	// these are DIAmeter-specific settings
```

Comment:
Not sure what you mean by DIameter specific settings. These are used in other places as well in the code - such as tide-search